### PR TITLE
[Workspace] Support workspace detail page

### PR DESCRIPTION
**Pull request by yubonluo**
### Description

Support workspace detail page
 - user can update workspace in the workspace detail
 - user can move to workspace detail page by click edit/workspace name in the workspace list page.
 - user can see get started card in the detail page (Subsequent updates)

### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/7240

## Screenshot
 - overview tab
image

 - settings tab
image

 - collaborators tab
image

## Testing the changes

 - Clone the latest code and run yarn osd bootstrap
 - Modify config/opensearch_dashboards.yml, add workspace.enabled: true
 - Run yarn start --no-base-path
 - Create a workspace and move to detail page

## Changelog
- feat: [Workspace] Support workspace detail page

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/public/components/workspace_detail/workspace_updater.tsx [hunk 1]
```diff
@@ -4,30 +4,29 @@
...
+  WorkspaceDetailForm,
 } from '../workspace_form';
 import { getDataSourcesList } from '../../utils';
 import { DataSource } from '../../../common/types';
 
 export interface WorkspaceUpdaterProps {
   workspaceConfigurableApps$?: BehaviorSubject<PublicAppInfo[]>;
-  hideTitle?: boolean;
-  maxWidth?: number | string;
+  tab?: string;
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/components/workspace_detail/workspace_updater.tsx:29]: Should be `DetailTab if I understood the props here correctly?

File: src/plugins/workspace/public/plugin.ts [hunk 2]
```diff
@@ -275,32 +274,17 @@ export class WorkspacePlugin implements Plugin<{}, {}, WorkspacePluginSetupDeps>
     });
 
     /**
-     * register workspace overview page
+     * register workspace detail page
      */
     core.application.register({
-      id: WORKSPACE_OVERVIEW_APP_ID,
+      id: WORKSPACE_DETAIL_APP_ID,
       title: i18n.translate('workspace.settings.workspaceOverview', {
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/plugin.ts:281]: Better to change the translate key as well.

File: src/plugins/workspace/public/components/workspace_overview/workspace_overview.tsx
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/components/workspace_overview/workspace_overview.tsx:1]: I thought we will still have the getting started sections in the workspace detail page.

**Comment by SuZhou-Joe** [comment]
From the screenshot: 
image

Don't you think the width is too short for displaying the description of the use case?

File: src/plugins/workspace/public/plugin.ts (hunk 2 above)
Comment (yubonluo) [reply src/plugins/workspace/public/plugin.ts]: Sure, it has been updated

File: src/plugins/workspace/public/components/workspace_detail/workspace_updater.tsx (hunk 1 above)
Comment (yubonluo) [reply src/plugins/workspace/public/components/workspace_detail/workspace_updater.tsx]: Sure, it has been updated.

File: src/plugins/workspace/public/components/workspace_overview/workspace_overview.tsx
Comment (yubonluo) [reply src/plugins/workspace/public/components/workspace_overview/workspace_overview.tsx:1]: I have not the final design about the getting started sections currently.

**Comment by yubonluo** [comment]
> From the screenshot: image
> 
> Don't you think the width is too short for displaying the description of the use case?
Sure, it has been updated

File: src/plugins/workspace/public/components/workspace_detail/workspace_updater.tsx [hunk 3]
```diff
@@ -4,30 +4,29 @@
...
+  WorkspaceDetailForm,
 } from '../workspace_form';
 import { getDataSourcesList } from '../../utils';
 import { DataSource } from '../../../common/types';
 
 export interface WorkspaceUpdaterProps {
   workspaceConfigurableApps$?: BehaviorSubject<PublicAppInfo[]>;
-  hideTitle?: boolean;
-  maxWidth?: number | string;
+  detailTab?: string;
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/components/workspace_detail/workspace_updater.tsx:29]: Should we use the enum instead of string?

**Review by SuZhou-Joe — APPROVED** [review]

File: src/plugins/workspace/public/components/workspace_detail/workspace_updater.tsx (hunk 3 above)
Comment (yubonluo) [reply src/plugins/workspace/public/components/workspace_detail/workspace_updater.tsx]: updated

**Review by SuZhou-Joe — APPROVED** [review]

File: src/plugins/workspace/public/components/workspace_detail/workspace_detail.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import { i18n } from '@osd/i18n';
+import { CoreStart, PublicAppInfo } from 'opensearch-dashboards/public';
+import { BehaviorSubject } from 'rxjs';
+import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
+import { WorkspaceDetailContent } from './workspace_detail_content';
+import { WorkspaceUpdater } from './workspace_updater';
+import { DetailTab } from '../workspace_form/constants';
+
+export interface WorkspaceDetailProps {
+  workspaceConfigurableApps$?: BehaviorSubject<PublicAppInfo[]>;
```
Comment (ruanyl) [inline src/plugins/workspace/public/components/workspace_detail/workspace_detail.tsx:27]: I had an impression that @wanglam removed `workspaceConfigurableApps$`, @wanglam can you have a look at this?

**Review by ruanyl — APPROVED** [review]

**Pull request merged**